libbeat/autodiscover/template/config.go

```diff
@@ -91,7 +112,7 @@ func (c Mapper) GetConfig(event bus.Event) []*common.Config {
 }
 
 // ApplyConfigTemplate takes a set of templated configs and applys information in an event map
-func ApplyConfigTemplate(event bus.Event, configs []*common.Config, keystoreEnabled bool) []*common.Config {
+func ApplyConfigTemplate(event bus.Event, configs []*common.Config, options []ucfg.Option) []*common.Config {
```

Comment:
nit: use `options ...ucfg.Option`.

+1 on removing the bool.

Do you have other use-cases besinds the resolver in mind? Maybe it makes sense to accept a 'Resolver' interface instead of all possible options?

Reply:
👍  on `options ...ucfg.Option`

About `Resolver`, I don't have in mind other cases right now, however I think that using just `options` there is ok and introducing a resolver might make things more complex and difficult in the future, but let me know what you think about this.

Reply:
👍